File: 25 November 2022/C++/Shreyansh and his bits.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <iostream>
using namespace std;


// } Driver Code Ends
//User function Template for C++

const int kN = 40;
long long cache[kN][kN];
class Solution{
public:
    
    long long ncr(int n, int r) {
        if (r > n) {
            return 0;
        }
        if (r == 0 || r == n) {
            return 1;
        }
        if (cache[n][r] != -1) {
            return cache[n][r];
        }
        
        auto res = ncr(n-1, r-1) + ncr(n-1, r);
        return (cache[n][r] = res);
    }
    long long count(long long x) {
        for (int i = 0; i < kN; i++) {
            for (int j = 0; j < kN; j++) {
                cache[i][j] = -1;
            }
        }
        long long res = 0;
        int ones = 0;
        int bits = 0;
        while (x != 0) {
            if (x&1) {
                res += ncr(bits, ++ones);
            }
            bits++;
            x >>= 1;
        }
        return res;
    }
};

//{ Driver Code Starts.

int main() {
    
	int t;
	cin >> t;
	while (t-- > 0) {
	    long long x; cin >> x;
	    Solution ob;
	    cout << ob.count(x) << '\n';
	}
	return 0;
}
// } Driver Code Ends

File: 25 November 2022/Python/Shreyansh and his bits.py
class Solution:
    def ncr(self,n,r):
        if r>n:
            return 0
        if r==0 or r==n:
            return 1
        if self.cache[n][r]!=-1:
            return self.cache[n][r]
        res=self.ncr(n-1,r-1)+self.ncr(n-1,r)
        self.cache[n][r]=res
        return res
    def count (self, N):
        self.cache=[[-1]*(40) for _ in range(40)]
        res=0
        ones=0
        bits=0
        while N!=0:
            if N&1:
                res+=self.ncr(bits,ones+1)
                ones+=1
            bits+=1
            N>>=1
        return res

#{ 
 # Driver Code Starts
if __name__ == '__main__': 
    ob = Solution()
    t = int (input ())
    for _ in range (t):
        N = int(input())
        print(ob.count(N))



# } Driver Code Ends